edgedb-tokio/src/tutorial.md

```diff
@@ -0,0 +1,725 @@
+# EdgeDB Rust client tutorial
+
+# Getting started (project template repo)
+
+If you just want a working repo to clone to get started, [go here](https://github.com/Dhghomon/edgedb_rust_client_examples), type `edgedb project init` to start an EdgeDB project and then `cargo run` to run the samples.
+
+This tutorial is essentially a more detailed version of the main.rs file inside that repo. It also uses the same schema as that in the EdgeDB [tutorial](https://www.edgedb.com/tutorial), plus a few extra types on top.
+
+# Getting started (detailed)
+
+## Cargo
+
+The minimum to add to your Cargo.toml to use the client is [edgedb-tokio](https://docs.rs/edgedb-tokio/latest/edgedb_tokio/):
+
+    edgedb-tokio = "0.3.0"
+
+The next most common dependency is [edgedb-protocol](https://docs.rs/edgedb-protocol/latest/edgedb_protocol/), which includes the EdgeDB types used for data modeling:
+
+    edgedb-protocol = "0.4.0"
+
+A third crate called [edgedb-derive](https://docs.rs/edgedb-derive/latest/edgedb_derive/) contains a `#[derive(Queryable)]` derive macro:
+
+    edgedb-derive = "0.4.0"
+    
+The Rust client uses tokio so add this to Cargo.toml as well:
+
+    tokio = { version = "1.27.0", features = ["macros", "rt-multi-thread"] }`
+
+If you are avoiding async code and want to emulate a blocking client, you will still need to use tokio as a dependency but can bridge with async using [one of the methods](https://tokio.rs/tokio/topics/bridging) recommended by tokio. This won't require any added features:
+
+    tokio = "1.27.0"
+
+And then you can start a runtime upon which you can use the `.block_on()` method to block and wait for futures to resolve. e.g.:
+
+```rust
+let rt = tokio::runtime::Builder::new_current_thread()
+    .enable_all()
+    .build()?;
+```
+
+## Edgedb CLI
+
+The `edgedb` CLI initializes EdgeDB projects in the same way cargo does, except it does not create a new directory. So to start a project, use `cargo new (your crate name)` as usual, then go into the directory and type `edgedb project init`. The CLI will prompt you for the instance name and version of EdgeDB to use. It will look something like this:
```

Comment:
Yes. This is a bit of a hint to my past self as when I first gave EdgeDB Projects a try in March my Cargo habit just led me to assume that it would create a project in a new directory. That in and of itself wasn't a big deal but I next made a Cargo project in a new directory and then `edgedb project init`ed inside it which confused the EdgeDB CLI by the edgedb.toml I hadn't deleted in the parent directory.